ctapipe/reco/shower_processor.py

```diff
@@ -70,21 +76,23 @@ def __call__(self, event: ArrayEventContainer):
         event : ctapipe.containers.ArrayEventContainer
             Top-level container for all event information.
         """
-        k = self.reconstructor_type
-        event.dl2.stereo.geometry[k] = self.reconstructor(event)
-
-        # compute and store the impact parameter for each reconstruction (for
-        # now there is only one, but in the future this should be a loop over
-        # reconstructors)
+        for reco_type, reconstructor in zip(
+            self.reconstructor_types, self.reconstructors
+        ):
+            reconstructor(event)
 
-        # for the stereo reconstructor:
-        impact_distances = shower_impact_distance(
-            shower_geom=event.dl2.stereo.geometry[k], subarray=self.subarray
-        )
+            # compute and store the impact parameter for each reconstruction
 
-        for tel_id in event.trigger.tels_with_trigger:
-            tel_index = self.subarray.tel_indices[tel_id]
-            event.dl2.tel[tel_id].impact[k] = TelescopeImpactParameterContainer(
-                distance=impact_distances[tel_index],
-                prefix=f"{self.reconstructor_type}_tel",
+            # for the stereo reconstructor:
+            impact_distances = shower_impact_distance(
```

Comment:
I think we should move this calculation into the stereo reconstructors or at least check if the reconstructor actually filled the geometry container.

Reply:
I moved it to `Reconstructor`